proposals/proposaldb_test.go

```diff
@@ -2,176 +2,155 @@ package proposals
 
 import (
 	"context"
-	"errors"
 	"testing"
 
 	"github.com/golang/mock/gomock"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/spacemeshos/go-spacemesh/codec"
 	"github.com/spacemeshos/go-spacemesh/common/types"
 	"github.com/spacemeshos/go-spacemesh/database"
 	"github.com/spacemeshos/go-spacemesh/log/logtest"
-	"github.com/spacemeshos/go-spacemesh/proposals/mocks"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+	"github.com/spacemeshos/go-spacemesh/mesh/mocks"
+	"github.com/spacemeshos/go-spacemesh/sql"
 )
 
 type testDB struct {
 	*DB
-	mockMesh *mocks.MockmeshDB
+	mesh *mesh.Mesh
 }
 
 func createTestDB(t *testing.T) *testDB {
 	types.SetLayersPerEpoch(layersPerEpoch)
+
+	db := sql.InMemory()
+	mdb, err := mesh.NewPersistentMeshDB(db, 1, logtest.New(t))
```

Comment:
why do you want to test querying proposals with ballots?
once proposals come in via gossip or sync, proposals and ballots are totally separate entities.
right now in tests Proposal is stitched back together with a separate Ballot obj. i am not sure if thats even necessary.
so yes, please dont use a real mesh in test

Reply:
I thought it would be better to just copy the original behavior. But assuming what you said, a real mesh is not needed, I've already removed it and will update the PR soon

Reply:
i think you are talking about the fetch, right? you should follow the other tables (ballots/atx) that serve hash requests with a blob

Reply:
in this cause you should do 

td.mockMesh.EXPECT().AddBallot(&p.Ballot).DoAndReturn(func() {
  // save ballots to db
}).Times(1)